controllers/progresses.js

```diff
@@ -107,8 +108,38 @@ const createProgress = async (req, res) => {
  */
 
 const getProgress = async (req, res) => {
+  const { dev, page = 0, size = 100, type, userId, taskId } = req.query;
   try {
     const data = await getProgressDocument(req.query);
+    if (dev) {
+      const paginatedProgressDocs = await getPaginatedProgressDocument(req.query);
+      const limit = parseInt(size, 10);
+      const offset = parseInt(page, 10) * limit;
+      const total = data.length;
+      const nextPage = offset + limit < total ? parseInt(page, 10) + 1 : null;
+      const prevPage = page > 0 ? parseInt(page, 10) - 1 : null;
+
+      let baseUrl = `${req.baseUrl}`;
+      if (type) {
+        baseUrl += `?type=${type}`;
+      } else if (userId) {
+        baseUrl += `?userId=${userId}`;
+      } else if (taskId) {
+        baseUrl += `?taskId=${taskId}`;
+      }
+      const nextLink = nextPage !== null ? `${baseUrl}&page=${nextPage}&size=${size}&dev=${dev}` : null;
+      const prevLink = prevPage !== null ? `${baseUrl}&page=${prevPage}&size=${size}&dev=${dev}` : null;
```

Comment:
```suggestion
      const nextLink = nextPage ? `${baseUrl}&page=${nextPage}&size=${size}&dev=${dev}` : null;
      const prevLink = prevPage ? `${baseUrl}&page=${prevPage}&size=${size}&dev=${dev}` : null;
```
Can we do it like this because undefined and null in javascript are falsy values.

Reply:
yes we can do it like this , thanks

Reply:
resolved

Reply:
actually we can not do this as prev page can be 0 and 0 will be considered falsy and that what we don't want